.github/workflows/workflow.yaml

```diff
@@ -10,37 +10,39 @@ defaults:
     shell: bash -l {0}
 
 jobs:
-  flake8:
-    timeout-minutes: 40
+  pre-commit-checks:
+    name: Pre-commit Checks
+    timeout-minutes: 30
     runs-on: ubuntu-latest
-
     steps:
-      - name: Check out code
-        uses: actions/checkout@v3
-
-      - name: Setup mamba
-        uses: ./.github/actions/mamba
-
-      - name: Flake8
-        run: |
-          micromamba run -n modyn flake8 --version
-          micromamba run -n modyn flake8 modyn --statistics
+      - name: Checkout branch
+        uses: actions/checkout@v4
+      - name: Run pre-commit-conda
+        uses: quantco/pre-commit-conda@v1
```

Comment:
Any reason for not using the mamba/action.yml that we use for the other jobs?

Reply:
The local action does some things we don't need for this use case.

Reply:
Similar to my other comment, it caches the environment though, doesn't it? And that speeds up the action

Reply:
The action runs in <30 sec. We don't need to optimize this further.